core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
I like the idea of something like `spark.execution.memoryFraction` to describe the fraction of memory that is reserved for internal use by Spark's own execution operators and not by caching or user-code.

As part of this renaming, I'd also like to rename ShuffleMemoryManager and consolidate it with Tungsten's new TaskMemoryManager and ExecutorMemoryManager in order to clarify the fact that ShuffleMemoryManager controls memory reservations for many different internal operators, many of which have nothing to do with shuffle.